### PR TITLE
GitAPI v0.1

### PR DESCRIPTION
**Pull request by JonasLukasczyk**

### CODE REVIEW

**Comment by JonasLukasczyk** [comment]
This PR adds the command `git update -r URLtoRemoteRepo -m CommitMsg`, where the -r and -m parameters are optional. This command is simply a convenience function that automatically performs a sequence of git operations to set the state of the git repo to the current state of the local arc folder. The command follows the following logic:

1. Iterate over all files in the repo and stage them either with `git add` or `git lfs track` based on a currently fixed size limit
2. Stage all removed files via `git add -u`
3. Perform a commit with the message passed via `-m` which defaults to "Update"
4. If `-r URLtoRemoteRepo` is present then perform `remote remove origin; remote add origin URLtoRemoteRepo`
5. If `remote origin` is defined then perform `git pull --rebase origin master`
6. If `remote origin` is defined then Perform `git push origin master`

Future improvements:
1. Step 1 can use an lfs filesize limit that has been specified in a config file.
2. Step 5 can be extended to handle merge conflicts via simple `(yes|no|abort)` options, where `yes` replaces all repo changes with local changes, `no` replaces all local changes with the repo state, and `abort` resets the local state to the last local commit.

File: src/ArcCommander/APIs/GitAPI.fs [hunk 1]
```diff
@@ -1,18 +1,94 @@
...
+        let trackWithLFS (file:string) =
+            if Fake.Tools.Git.CommandHelper.directRunGitCommand repoDir ("lfs track "+file)
+            then printfn "git lfs track %s" file; trackWithAdd (repoDir+".gitattributes")
+            else printfn "[ERROR] git lfs track %s" file
+
+        allFilesPlusSizes |> List.iter(
+            fun pair ->
+                let (file,size) = pair
+
+                if size>150000000L
```
Comment (cgarth) [inline src/ArcCommander/APIs/GitAPI.fs:54]: should be a config option

File: src/ArcCommander/APIs/GitAPI.fs [hunk 2]
```diff
@@ -1,18 +1,94 @@
...
+            | Some s -> s
+        // printfn "%A" (Fake.Tools.Git.CommandHelper.runGitCommand repoDir ("status"))
+        printfn "commit -m '%s'" commitMessage
+        Fake.Tools.Git.Commit.exec repoDir commitMessage |> ignore
+
+        // add remote if specified
+        match tryGetFieldValueByName "RepositoryAdress" gitArgs with
+            | Some "" | None -> ()
+            | Some remote ->
+                Fake.Tools.Git.CommandHelper.directRunGitCommand repoDir ("remote remove origin") |> ignore
```
Comment (cgarth) [inline src/ArcCommander/APIs/GitAPI.fs:75]: factor into `git` wrapper function that also does error reporting

File: src/ArcCommander/APIs/GitAPI.fs [hunk 3]
```diff
@@ -1,18 +1,94 @@
...
-    // ToDo : Implement
-    let update (arcConfiguration:ArcConfiguration) (gitArgs:Map<string,Argument>) = 
-        raise (NotImplementedException())
-        |> ignore
+    let getRepoDir (arcConfiguration:ArcConfiguration) =
+        let workdir = GeneralConfiguration.getWorkDirectory arcConfiguration
+        printfn "%s" workdir
+
+        let gitDir = Fake.Tools.Git.CommandHelper.findGitDir(workdir).FullName
+        gitDir.Substring(0,gitDir.Length-4)
```
Comment (cgarth) [inline src/ArcCommander/APIs/GitAPI.fs:15]: use Path.getDirectory

**Review by cgarth — COMMENTED** [review]
whitespace changes?

**Comment by JonasLukasczyk** [comment]
Now that we decided to go this route I cleaned up the code.

File: src/ArcCommander/APIs/GitAPI.fs (hunk 1 above)
Comment (HLWeil) [reply src/ArcCommander/APIs/GitAPI.fs:58]: Agreed. @JonasLukasczyk would you add this? 

You could just add an according function [here](
https://github.com/nfdi4plants/arcCommander/blob/b92dacb949dc69a3b9fa0b38beaa34f714a16291/src/ArcCommander/ArcConfiguration.fs#L85)

And then set the default value with the according key in the [config file](https://github.com/nfdi4plants/arcCommander/blob/developer/src/ArcCommander/config)

Alternatively I could do this after merging your PR

**Comment by HLWeil** [comment]
Hey, sry for the long delay. This looks very good!

I just commented about the git lfs file size threshold. Otherwise the PR is fine to be merged.

**Pull request merged**